tests/test_cli.py

```diff
@@ -0,0 +1,213 @@
+            )
+        else:  # command fails for another reason i.e. a test failure
+            pytest.fail(
+                "The browser is available but the command to fetch "
+                "the history from it has failed."
+            )
+
+
+def test_format_argument():
+    """Tests arguments for the format option."""
+    # First check format of default:
+    csv_output = subprocess.check_output([CMD_ROOT]).decode("utf-8")
+    # Sniffer determines format, less intensive than reading in csv.reader
+    # and we don't mind the CSV dialect, so just check call doesn't error
+    read_csv = csv.Sniffer()
+    # This gives '_csv.Error: Could not determine delimiter' if not a csv file
+    read_csv.sniff(csv_output, delimiters=",")
+    assert read_csv.has_header(csv_output)
+
+    for fmt_opt in VALID_CMD_OPTS[3]:
+        for fmt_arg in VALID_FORMAT_ARGS:
+            output = subprocess.check_output([CMD_ROOT, fmt_opt, fmt_arg]).decode(
+                "utf-8"
+            )
+            if fmt_arg in ("csv", "infer"):  # infer gives csv if no file
+                read_csv.sniff(output, delimiters=",")
+                assert read_csv.has_header(output)
+                assert CSV_HISTORY_HEADER in output
+            elif fmt_arg == "json":
+                assert output.startswith(HISTORY_TITLE)
+                # Need to strip header off before checking for valid json
+                json.loads(output.lstrip(HISTORY_TITLE))
+            else:  # i.e. fmt_arg == 'jsonl':
+                assert output.startswith(HISTORY_TITLE)
+                jsonl_output = output.lstrip(HISTORY_TITLE)
+                # Newline-delimited json so test each line is valid json
+                for line in jsonl_output.splitlines():
+                    json.loads(line)
+
+
+def test_output_argument():
+    """Test arguments for the output option."""
+    for output_opt in VALID_CMD_OPTS[4]:
+        with tempfile.NamedTemporaryFile(suffix=".csv") as f:
+            output = subprocess.check_output([CMD_ROOT, output_opt, f.name])
+            # Check output was not sent to STDOUT since should go to file
+            assert HISTORY_HEADER not in output.decode("utf-8")
+            with open(f.name, "rb") as f:  # now check the file
+                assert HISTORY_HEADER in f.read().decode("utf-8")
+
+
+def test_argument_combinations():
+    """Test that combinations of optional arguments work properly."""
+    # Choose some representative combinations. No need to test every one.
+    indices = (0, 1)
+    # To test combos of short or long option variants
+    for index_a, index_b in itertools.product(indices, indices):
+        chrome_bookmarks_output = subprocess.check_output(
+            [
+                CMD_ROOT,
+                VALID_CMD_OPTS[1][index_a],  # type
+                VALID_TYPE_ARGS[1],  # ... is bookmarks,
+                VALID_CMD_OPTS[2][index_a],  # browser
+                VALID_BROWSER_ARGS[1],  # ... is Chrome
```

Comment:
Ah true, good spot. I'll make sure it is handled appropriately. I'll register a browser that is previously determined to be installed and use that instead of always trying Chrome. That way it is only skipped in the case no browsers are installed, which should be very unlikely for any tester.